### PR TITLE
Elasticsearch5 sink

### PR DESCRIPTION
**Pull request by geoffroyvergne**
Hi Flume team,
I worked for my company on this patch to be able to use ElasticSearch 5 with flume.
We heavy use Flume with Hadoop and ElasticSearch as well.
We are testing this patch that seams to work properly
I thought it could be helpfull for some of your users

Let me know if you have any questions

Geoffroy

### CODE REVIEW

**Comment by liorze** [comment]
Hi @geoffroyvergne, thank you for your contribution.

Can you please clean up your code a bit? i.e. minimize the changeset (remove comments and whitespace changes).

File: flume-ng-sinks/flume-ng-elasticsearch-sink/src/test/java/org/apache/flume/sink/elasticsearch/TestElasticSearchIndexRequestBuilderFactory.java [hunk 1]
```diff
@@ -24,46 +24,63 @@
...
       }
     };
   }
 
   @Test
   public void shouldUseUtcAsBasisForDateFormat() {
-    assertEquals("Coordinated Universal Time",
-        factory.fastDateFormat.getTimeZone().getDisplayName());
+    String timeZone = factory.fastDateFormat.getTimeZone().getDisplayName();
+    assertTrue("Temps universel coordonné".equals(timeZone) ||
```
Comment (zouyonghao) [inline flume-ng-sinks/flume-ng-elasticsearch-sink/src/test/java/org/apache/flume/sink/elasticsearch/TestElasticSearchIndexRequestBuilderFactory.java]: I think here you should use
```java
assertEquals("Coordinated Universal Time",
        factory.fastDateFormat.getTimeZone().getDisplayName(Locale.ENGLISH));
```

And additionly, I'm not sure whether you see my email.

**Comment by zouyonghao** [comment]
I think here you should use
```java
assertEquals("Coordinated Universal Time",
        factory.fastDateFormat.getTimeZone().getDisplayName(Locale.ENGLISH));
```

in the ```TestElasticSearchIndexRequestBuilderFactory.java```

And additionly, I'm not sure whether you see my email that replied in the dev maillist.

**Comment by zouyonghao** [comment]
I put my patch for this on
https://issues.apache.org/jira/browse/FLUME-3021

I fixed all unit tests and it should work. But need more tests.
And it seems it need some configuration files like some in your pr, you can feel free to test this patch and add your code to this.

Thanks.

**Comment by geoffroyvergne** [comment]
Hi there,
As you advised, I clean code (remove unusedd comments, and white spaces)
And i fixed timezones assertions in the tests

BR

**Comment by geoffroyvergne** [comment]
Hi there,
As you advised, I clean code (remove unusedd comments, and white spaces)
And i fixed timezones assertions in the tests

BR

**Comment by liorze** [comment]
Thanks. Can you please also enable the checkstyle plugin (uncomment it in the pom file)?

We will also have to merge this one first https://reviews.apache.org/r/53056/, as elasticsearch 5.x requires Java 8.

**Comment by zouyonghao** [comment]
Hi,
It seem all tests in ```TestElasticSearchSink``` could not be passed bacause the ```createNodes``` method do nothing. The solution is trying to mock the node.
You can take a look at https://reviews.apache.org/r/57538/
I fixed all tests.
Or maybe you can merge my commits at https://github.com/zouyonghao/flume
the trunk branch. 
But I'm not sure whether to create a new pr or just use your pr, and whether you should post your patch at jira or just use github. I'm just a newbee to open source project.

Thanks.

**Comment by geoffroyvergne** [comment]
Hi, 
If you want to pass all tests, you need to have an ElasticSearch instance running in local.
The standalone node doesn't exists anymore on the release 5.x
There is a maven dependency test


<dependency>
    <groupId>org.elasticsearch.test</groupId>
    <artifactId>framework</artifactId>
    <version>5.2.1</version>
</dependency>

But I didn't implement it yet
I'll try also to take a look at your patch

cheers

**Comment by zouyonghao** [comment]
But the unit tests are not integration tests, should be passed without any dependencies.
And the embbed server has removed in es 5.x.
So in my patch, I mock a node that can be used by the ```openLocalDiscoveryClient``` method which is defined in ```ElasticSearchTransportClient```.

**Comment by bearrito** [comment]
How does this compare to https://issues.apache.org/jira/browse/FLUME-3021

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by rgoers** [comment]
Support for Elasticsearch has been removed from the main Flume build since it is so out of date and its new license is problematic. Please work against the flume-search repo.

**Pull request closed without merging**